### DISCUSSION TITLE
Individual grain labeling

### DISCUSSION THREAD

**Discussion by RafaelM-pixel** [mtex-toolbox/mtex#1699] [Ask Anything]
Hi everyone,
First of all I would like to thank you for the great job you do to help novice researchers.
I am not very well versed in MTEX nor in MATLAB, but before coming here I tried with the program and searched through previous questions without success.
I have an small scanned area, and I would like to label each grain with the average orientation with Miller indices considering the plane normal to the electron beam, I would appreciate your suggestions and help.

**Comment by kilir** [comment]
Hi,
how about something like that?
```
mtexdata titanium
%%
grains  = ebsd.calcGrains
grains = grains(grains.grainSize>50)
%%
h = inv(grains.meanOrientation) .* vector3d(1,2,3) % put something useful here

plot(grains)
for i=1:length(grains)
text(grains(i),char(h(i)))
end
```

Cheers,
Rüdiger

**Reply by RafaelM-pixel** [reply]
Hi Rüdiger,
Thank you so much for your answer, it helps me a lot.

Take care and stay safe
